apps/mobile-wallet/src/features/modals/BottomModalFlashList.tsx

```diff
@@ -0,0 +1,170 @@
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+import { ContentStyle } from '@shopify/flash-list'
+import { ReactNode } from 'react'
+import { KeyboardAvoidingView, Pressable } from 'react-native'
+import { GestureDetector } from 'react-native-gesture-handler'
+import Animated from 'react-native-reanimated'
+import styled from 'styled-components/native'
+
+import AppText from '~/components/AppText'
+import { CloseButton } from '~/components/buttons/Button'
+import { ContentScrollHandlers, useBottomModalState } from '~/features/modals/useBottomModalState'
+import { DEFAULT_MARGIN } from '~/style/globalStyle'
+
+export interface BottomModalFlashListProps {
+  modalId: number
+  onClose?: () => void
+  title?: string
+  maximisedContent?: boolean
+  minHeight?: number
+  navHeight?: number
+  noPadding?: boolean
+  children: (scrollHandlers: {
+    contentContainerStyle: ContentStyle
+    onScroll: ContentScrollHandlers['onScroll']
+    onScrollBeginDrag: ContentScrollHandlers['onScrollBeginDrag']
+    onScrollEndDrag: ContentScrollHandlers['onScrollEndDrag']
+    onContentSizeChange: (w: number, h: number) => void
+  }) => ReactNode
+}
+
+const AnimatedPressable = Animated.createAnimatedComponent(Pressable)
+
+const BottomModalFlashList = ({
+  modalId,
+  onClose,
+  title,
+  maximisedContent,
+  minHeight,
+  navHeight = 50,
+  noPadding,
+  children
+}: BottomModalFlashListProps) => {
+  const {
+    modalHeightAnimatedStyle,
+    handleAnimatedStyle,
+    backdropAnimatedStyle,
+    handleContentSizeChange,
+    panGesture,
+    handleClose,
+    contentScrollHandlers
+  } = useBottomModalState({
+    modalId,
+    maximisedContent,
+    minHeight,
+    navHeight,
+    onClose
+  })
+
+  return (
+    <GestureDetector gesture={panGesture}>
+      <KeyboardAvoidingViewStyled behavior="height" enabled={!maximisedContent}>
+        <Backdrop style={backdropAnimatedStyle} onPress={handleClose} />
+        <Container>
+          <ModalStyled style={modalHeightAnimatedStyle}>
+            <HandleContainer>
+              <Handle style={handleAnimatedStyle} />
+            </HandleContainer>
+            <Navigation style={{ height: navHeight }}>
+              <NavigationButtonContainer align="left" />
+              <Title semiBold>{title}</Title>
+              <NavigationButtonContainer align="right">
+                <CloseButton onPress={handleClose} compact />
+              </NavigationButtonContainer>
+            </Navigation>
+            {children({
+              ...contentScrollHandlers,
+              onContentSizeChange: handleContentSizeChange,
+              contentContainerStyle: {
+                padding: noPadding ? 0 : DEFAULT_MARGIN
+              }
+            })}
+          </ModalStyled>
+        </Container>
+      </KeyboardAvoidingViewStyled>
+    </GestureDetector>
+  )
+}
+
+export default BottomModalFlashList
+
+const KeyboardAvoidingViewStyled = styled(KeyboardAvoidingView)`
```

Comment:
Styles are the same as in `BottomModal` for now. But it may change in the future - I want to refactor the modals header & title. The implementation will probably not be the same in both BottomModal variants.